hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkFileFormatInternalRowReaderContext.scala

```diff
@@ -78,16 +80,16 @@ class SparkFileFormatInternalRowReaderContext(baseFileReader: SparkColumnarFileR
       assert(getRecordContext.supportsParquetRowIndex())
     }
     val structType = HoodieInternalRowUtils.getCachedSchema(requiredSchema)
+    val (readSchema, readFilters) = getSchemaAndFiltersForRead(structType, hasRowIndexField)
```

Comment:
can the filters be ensured to be based on pk? and what is the `requiredFilters` there?

Reply:
Now it can be confirmed that it is so, because currently only the mor logic will reach here. For mor reading, the filter returned here must be related to the primary key